src/components/CardGenerator.js

```diff
@@ -22,13 +23,34 @@ class CardGenerator extends React.Component {
         email: '',
         linkedin: '',
         github: '',
-        photo: ''
+        photo: defaultImage
       },
-      isOpen: 1
+      isOpen: 1,
+      isAvatarDefault: true
     };
+
+    this.fr = new FileReader();
+    this.myFileField = React.createRef();
+
     this.handleInputChange = this.handleInputChange.bind(this);
     this.openPanel = this.openPanel.bind(this);
+    this.handleFilePicker = this.handleFilePicker.bind(this);
+    this.uploadImage = this.uploadImage.bind(this);
+    this.updateAvatar = this.updateAvatar.bind(this);
+    this.getImage = this.getImage.bind(this);
   }
+
+  updateAvatar(img) {
+    const { data } = this.state;
+    this.setState(prevState => {
+      const newProfile = { ...data, photo: img };
```

Comment:
we think instead of ...data, you should use prevState.data or you are using prevState as a parameter for nothing.